### PR TITLE
[CIS-858] Always refetch queries when reconnected

### PR DESCRIPTION
**Pull request by VojtaStavik**
Turn our the `/sync` endpoint returns only the events for the channels specified in the request. This means if the current user is added to a new channel while offline, the new channel won't appear in the channel list.

- the first commit is just renaming, clean-up, and reorganizing of existing api
- the second one is the actual functionality

### CODE REVIEW

File: Sources/StreamChat/Workers/DatabaseCleanupUpdater.swift [hunk 1]
```diff
@@ -31,29 +31,23 @@ class ChannelDatabaseCleanupUpdater<ExtraData: ExtraDataTypes>: Worker {
...
+    ///
+    /// - Parameter session: session for writing into the database.
+    ///
+    func resetExistingChannelsData(session: DatabaseSession) throws {
         if let channels = try (session as? NSManagedObjectContext)?
             .fetch(ChannelDTO.allChannelsFetchRequest) {
-            channels.forEach { $0.resetLocalData() }
+            channels.forEach {
+                $0.resetLocalData()
+                print($0.cid)
```
Comment (b-onc) [inline Sources/StreamChat/Workers/DatabaseCleanupUpdater.swift:44]: forgot?

**Review by b-onc — APPROVED** [review]
:shipit:

File: Sources/StreamChat/Workers/DatabaseCleanupUpdater.swift (hunk 1 above)
Comment (VojtaStavik) [reply Sources/StreamChat/Workers/DatabaseCleanupUpdater.swift:44]: Yes 🙈

Comment (VojtaStavik) [reply Sources/StreamChat/Workers/DatabaseCleanupUpdater.swift]: 57a8263

File: Sources/StreamChat/Workers/DatabaseCleanupUpdater_Tests.swift [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
-final class ChannelListCleanupUpdater_Tests: StressTestCase {
+final class DatabaseCleanupUpdater_Tests: StressTestCase {
     typealias ExtraData = NoExtraData
         
     var database: DatabaseContainerMock!
     var webSocketClient: WebSocketClientMock!
     var apiClient: APIClientMock!
     
-    var channelListCleanupUpdater: ChannelDatabaseCleanupUpdater<ExtraData>?
+    var channelListCleanupUpdater: DatabaseCleanupUpdater<ExtraData>?
```
Comment (dmigach) [inline Sources/StreamChat/Workers/DatabaseCleanupUpdater_Tests.swift:16]: This property's name has dodged the renaming wave

**Review by dmigach — APPROVED** [review]

File: Sources/StreamChat/Workers/DatabaseCleanupUpdater_Tests.swift (hunk 2 above)
Comment (VojtaStavik) [reply Sources/StreamChat/Workers/DatabaseCleanupUpdater_Tests.swift:16]: Right, will fix. Thanks!

**Pull request merged**